### DISCUSSION TITLE
[QUESTION] How to check if a user block and stop bot

### DISCUSSION THREAD

**Discussion by bsimjoo** [python-telegram-bot/python-telegram-bot#2485] [Q&A] [answered]
### How to check if a user block and stop bot
I use a database to store all users ids and information, and then my bot iterate users id and send message to them. But when a user block and stop bot I don't know how to remove user from database and bot will try to send message to a blocked user and this will raise an exception. I added a handler of type `MessageHandler` with `Filters.update` to check all update and I saw my bot does not receive any update when a user block and stop bot. so is there any other way except handling the exception?

### Related part of your code
```python
def log_update(u: Update, c:CallbackContext):
    message = (
        'Received a new update event from telegram\n'
        f'update = {json.dumps(u.to_dict(), indent = 2, ensure_ascii = False)}\n'
        f'user_data = {json.dumps(c.user_data, indent = 2, ensure_ascii = False)}\n'
        f'chat_data = {json.dumps(c.chat_data, indent = 2, ensure_ascii = False)}'
    )
    logging.info(message)
    if self.debug:
        try:
            self.bot.send_message(self.ownerID,html.escape(message), parse_mode = ParseMode.HTML)
        except Exception as e:
            self.log_bug(e,'Exception while sending update log to owner',ownerID=self.ownerID, message=html.escape(message))

self.dispatcher.add_handler(MessageHandler(Filters.update,log_update))
```

**Comment by Bibo-Joshi** [comment, marked as answer]
As of API 5.1 you will get an update with `update.my_chat_member`, where `update.my_chat_member.new_chat_member.status` can be used to check if the bot has been blocked. In PTB v13.4+ you can use `ChatMemberHandler` to catch those updates.
Nevertherless, handling exceptions is a very sane & advisable thing to do. IIRC `Unauthorized`  is raised when the user has blocked the bot, but you can also check the exception message (`str(e)`) to see that.

**Comment by bsimjoo** [comment]
Can I use `ChatMemberHandler` to check if the bot is kicked out of a group or channel?

**Comment by Bibo-Joshi** [comment]
Yes. By comparing the `old/new_chat_member_status` you can infere what has happened
